### PR TITLE
Huge update for project

### PR DESCRIPTION
**Pull request by Shammad-Kosayev**
+Validations are brought up to Controller layer.
+Unit tests added
+i forgot what i have done since update is huge, lol

### CODE REVIEW

File: src/main/java/az/edu/strangers/dao/Family.java [hunk 1]
```diff
@@ -9,7 +9,7 @@ public class Family implements HumanCreator {
 
     private Human father;
     private Human mother;
-    private Pet pet;
+    private List<Pet> pets;
```
Comment (sabuhihuseynov) [inline src/main/java/az/edu/strangers/dao/Family.java]: should be Set

File: src/main/java/az/edu/strangers/controller/FamilyController.java [hunk 2]
```diff
@@ -14,15 +19,64 @@ public FamilyController(FamilyService familyService) {
...
+        if (family == null || child == null) return Optional.empty();
+
+        return familyService.adoptChild(family, child);
+    }
+
+    public void deleteAllChildrenOlderThen(final Integer age) {
+        if (age != null) familyService.deleteAllChildrenOlderThen(age);
+    }
+
+    public Optional<List<Pet>> getPets(final Integer index) {
```
Comment (sabuhihuseynov) [inline src/main/java/az/edu/strangers/controller/FamilyController.java]: if you return list no need Optional

**Review by sabuhihuseynov — CHANGES_REQUESTED** [review]
pls look

File: src/test/java/az/edu/strangers/service/FamilyServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class FamilyServiceImplTest {
+    private static final FamilyDao dao = new CollectionFamilyDao();
```
Comment (sabuhihuseynov) [inline src/test/java/az/edu/strangers/service/FamilyServiceImplTest.java]: why static?

File: src/main/java/az/edu/strangers/service/FamilyServiceImpl.java [hunk 4]
```diff
@@ -53,7 +54,9 @@ public List<Family> getFamiliesLessThan(Integer number) {
 
     @Override
     public Long countFamiliesWithMemberNumber(Integer number) {
-        return getAllFamilies().stream()
+        return familyDao.getAllFamilies()
+                .stream()
+                .peek(System.out::println)
```
Comment (sabuhihuseynov) [inline src/main/java/az/edu/strangers/service/FamilyServiceImpl.java]: why printing families inside this method?

**Review by sabuhihuseynov — CHANGES_REQUESTED** [review]
pls look

File: src/main/java/az/edu/strangers/service/FamilyServiceImpl.java (hunk 4 above)
Comment (Shammad-Kosayev) [reply src/main/java/az/edu/strangers/service/FamilyServiceImpl.java]: i forgot to remove that lol

**Review by ayten-github — APPROVED** [review]

**Review by ayten-github — APPROVED** [review]

**Comment by ayten-github** [comment]
problem related to merging

**Pull request closed without merging**